CabalHelper/Wrapper.hs

```diff
@@ -142,13 +142,18 @@ main = handlePanic $ do
 \Cabal version %s was requested setup configuration was\n\
 \written by version %s" (showVersion ver) (showVersion hdrCabalVersion)
             _ -> do
-              eexe <- compileHelper opts hdrCabalVersion projdir distdir
-              case eexe of
-                  Left e -> exitWith e
-                  Right exe ->
-                    case args' of
-                      "print-exe":_ -> putStrLn exe
-                      _ -> do
-                        (_,_,_,h) <- createProcess $ proc exe args
-                        exitWith =<< waitForProcess h
+              case args' of
+                "print-exe-prepared":[] -> do
+                  isPrepared <- helperPrepared hdrCabalVersion
+                  putStrLn $ if isPrepared then "True" else "False"
```

Comment:
Come to think of it: I'd rather this command returned the status via executable exit code without printing anything. `is-prepared` is probably a more appropriate name then.